base-template.sh

```diff
@@ -98,6 +99,57 @@ set_main_variables() {
 "
 }
 
+############################################################
+# We register this repository for future potential
+# autoupdates if all of the following is true
+#   - core.hooksPath is not defined, meaning this hook
+#     needs to be in `.git/hooks`.
+#   - its not yet registered.
+#   - its a non-single install.
+#
+# Returns: 0
+############################################################
+register_for_autoupdate_if_needed() {
+    if ! git config --local githooks.autoupdate.registered >/dev/null 2>&1 &&
+        [ "$(git config --local githooks.single.install)" != "yes" ] &&
+        [ ! -d "$(git config --global core.hooksPath)" ]; then
+        register_repo_for_autoupdate "$CURRENT_GIT_DIR"
+    fi
+    return 0
```

Comment:
Nit: do we need the `return 0` in the functions where we don't actually care about the results?